tests/neg-custom-args/matchtype-loop.scala

```diff
@@ -6,9 +6,9 @@ object Test {
     case Int => LL[LL[X]]
   }
   def a: L[Boolean] = ???
-  def b: L[Int] = ???
+  // def b: L[Int] = ??? // times out
```

Comment:
Why do these time out now? Isn't that a problem?

Reply:
> Why do these time out now?

Because instead of stoping after one step of reduction we now try to keep reducing the match type body after the first step, which causes in infinite loop.

All the infinite loops that we encountered in match type reduction would eventually run out of stack, but this nicely trampolines and ultimately fails with an out of memory error.

This is indeed a problem, but the only solution I see would be to add some sort of reduction counter to match type reduction similar to `-Xmax-inlines`.